core/src/Streamly/Data/Serialize.hs

```diff
@@ -51,14 +51,20 @@ module Streamly.Data.Serialize
 
     -- * Serialize
     , Serialize(..)
-    -- XXX Need to export the config setters as well.
-    , defaultConfig
+
+    -- Deriving instances
+    , Config -- XXX rename to SerializeConfig
+    , defaultConfig -- XXX rename to defaultSerializeConfig
+    , inlineSize
+    , inlineSerialize
+    , inlineDeserialize
```

Comment:
Even `deriveSerialize` looks like a config option.
You can only distinguish `inlineSerialize` and `deriveSerialize` only by looking at the signature.